dra-evolution/pkg/api/claim_types.go

```diff
@@ -368,8 +292,22 @@ type ResourceRequestDetail struct {
 	Requirements []Requirement `json:"requirements,omitempty" protobuf:"bytes,4,opt,name=requirements"`
 }
 
-// CountDetails defines how many instances are desired.
-type CountDetails struct {
+// RequestClassReference must have one and only one field set.
+//
+// If we ever need to introduce new class settings which cannot be added to DeviceClass
+// (for example, because it wouldn't be okay for an older scheduler to ignore them),
+// then we could add new types and entries here to indicate to old schedulers that
+// they cannot handle the claim.
+type RequestClassReference struct {
+	// When referencing a DeviceClass, a request inherits additional
+	// configuration and requirements.
+	//
+	// +optional
+	DeviceClassName *string `json:"deviceClassName,omitempty"`
```

Comment:
So the idea here is that if we have something other than devices in the resource pool, you would have a different type of class here?

Reply:
That, or we add a more general concept than "device class" and then link to objects of that type here also for devices.

Reply:
I am *still* not buying the distinction between Resource and Device

Reply:
I removed `RequestClassReference`.